src/libstore/build/local-derivation-goal.cc

```diff
@@ -398,6 +398,19 @@ static void doBind(const Path & source, const Path & target, bool optional = fal
     if (stat(source.c_str(), &st) == -1) {
         if (optional && errno == ENOENT)
             return;
+        // EACCES can happen when a input path is a symlink to outside the store
+        else if (errno == EACCES) {
+            copyPath(source, target);
+            return;
+            // either link or copy
+            // link doesn't work at all
```

Comment:
Doesn't work in the case of the example in the commit message


Hmm
```
  indirect-symlink = pkgs.runCommand "indirect-symlink" { } ''
    cat ${direct-symlink}
    ln -vs ${direct-symlink} $out
  '';
```
`> cat: /nix/store/k3isri0ks6l63rk54z4714dc67sd1cp3-direct-symlink: No such file or directory`

Reply:
Commented out the auto deletion of the chroot dir

```
$ sudo ls -l /nix/store/q5glq907j6jn39aw99ncmcv5ijvnh5cs-indirect-symlink.drv.chroot/nix/store/55lg749likjz3ivq2varjcbj17xjam40-direct-symlink
lrwxrwxrwx root root 37 B Sat Jan 20 18:13:44 2024  /nix/store/q5glq907j6jn39aw99ncmcv5ijvnh5cs-indirect-symlink.drv.chroot/nix/store/55lg749likjz3ivq2varjcbj17xjam40-direct-symlink@ ⇒ /home/artturin/nixgits/my-nixpkgs/.version
```

I think what is happening is that the symlink to `/home/...` is being mounted/linked/copied to the chroot but the chroot doesn't have access to the result of the symlink

Reply:
I checked on 2.18.1 and there was still `cat: /nix/store/55lg749likjz3ivq2varjcbj17xjam40-direct-symlink: No such file or directory`, therefore it's not a regression caused by #8965